app/models/course.rb

```diff
@@ -6,4 +6,14 @@ class Course < ActiveRecord::Base
   belongs_to :creator, class_name: User.name
   has_many :course_users, inverse_of: :course, dependent: :destroy
   has_many :users, through: :course_users
+
+  after_initialize :set_defaults, if: :new_record?
```

Comment:
1. includes
 2. callbacks
 3. validations
 4. relations

if I read it correctly. @jsyeo?